### ISSUE TITLE
Add new order statuses and new label logic

### ISSUE THREAD

**Issue by bilfeldt** [smartsendio/woocommerce#7]
We would very much like to implement more automation into the plugin and this issue is meant to make that possible to implement in the future.

 # Statuses
The idea is that a WooCommerce order already can have the following [statuses](https://docs.woocommerce.com/document/managing-orders/):

- `Pending payment`: Order received, no payment initiated. Awaiting payment (unpaid).
- `Failed`: Payment failed or was declined (unpaid). Note that this status may not show immediately and instead show as Pending until verified (e.g., PayPal).
- `Processing`: Payment received (paid) and stock has been reduced; order is awaiting fulfillment. All product orders require processing, except those that only contain products which are both Virtual and Downloadable.
- `Completed`: Order fulfilled and complete – requires no further action.
- `On-Hold`: Awaiting payment – stock is reduced, but you need to confirm payment.
- `Cancelled`: Cancelled by an admin or the customer – stock is increased, no further action required.
- `Refunded`: Refunded by an admin – no further action required.

We are missing some steps between the `Processing` and `Completed` steps. In order to deal with fullfillment then we would like to add these statuses:

- `SmartSend-Pending`: Order is not yet sent to Smart Send for processing but has been queued and will soon be sent to Smart Send for processing.
- `SmartSend-Processing`: Smart Send accepted the request and is currently processing the order
- `SmartSend-Completed`: Smart Send has handled the order and a tracking number has been generated
- `SmartSend-Error`: There were an error with the handling request. Smart Send returned an error explaining what went wrong immidiatly when receiving the request.
- `SmartSend-Failed`: Smart Send accepted the request for processing byt the processing subsequently failed

We found inspiration for this from how Shopify does the fulfillment handling - see the field [status](https://help.shopify.com/en/api/reference/shipping-and-fulfillment/fulfillment)

# Manual handling of single order
Manual handling of a single order - when we are clicking _Generate label_ from the _Order View_ page. Here we skip the `SmartSend-Pending` status and instantly make an `instant processing` request. The immidiate response is either success or failure and hence the status is changed to `SmartSend-Completed` or `SmartSend-Error`

# Manual handling of less than five orders
This is when we select _Generate label_ from the _Order list_ page while having selected maximum 5 orders. Here we loop over the orders and process them immidiatly like in the _Manual handling of single order_ point above.

# Manual handling of more than five orders
This is when we select _Generate label_ from the _Order list_ page while having selected more than 5 orders. Here we make an API call containing all orders. We will get an instance response with a status for each order. We then loop over the orders and update them to either `SmartSend-Processing` (we succesfully queued the order for handling) or `SmartSend-Error` (something failed our validation for the order).

Our system will then handle the orders and make API requests to WooCommerce for each order once the order is handled. The order will be updated to either  `SmartSend-Completed` or `SmartSend-Failed` depending on the result.

**Question:** How should we save information about the error that occured?

# Automated handling
Automated handling is when the setup some kind of trigger which will automatically queue the order for handling. It might for example be a rule that once the order gets the status `Processing` then the order should be handled. Such rules can either be added to our plugin or they can be done by the webshop owner himself.

When the order status is changed to `SmartSend-Pending` then it's our cue to start the processing. We need to make an API request so that Smart Send can start handling the request. It would be best if the request could be async so that we do not slow down other processes. There are some WooCommerce [classes](https://docs.woocommerce.com/wc-apidocs/class-WP_Async_Request.html#methods) for that.

## Challenge
If WooCommerce has some good way of handling queues then it would make sense to simple add a WooCommerce queue job to make an API request to Smart Send and then handle the response right away. So making a request like the one in section _Manual handling of single order_

If WooCommerce does **not** have a good way of handling queues, then it would make sense to instantly when something is triggered make an API call to Smart Send but an async API call meaning an API call saying "please handle this order soon, but not now". That way the API call should be fast and Smart Send will then make an API call to WooCommerce once the order is handled.

**Basically the question is:** Should the queue handling be done at WooCommerce or at Smart Send?
